core/src/main/codegen/templates/Parser.jj

```diff
@@ -6182,6 +6182,12 @@ SqlTypeNameSpec DateTimeTypeName() :
     {
         return new SqlBasicTypeNameSpec(typeName, precision, s.end(this));
     }
+|
```

Comment:
is this supposed to be in Babel?

Reply:
Regarding Babel, my initial plan was to implement DATETIME datatype parsing at the Babel parser level, but I noticed that we have DATETIME literal parsing for BigQuery in the core, which is also valid for MySQL. Therefore, I decided to put DATETIME type parsing in Parser.jj, as it would be better to have both DATETIME datatype parsing and DATETIME literal parsing in the same place.